eWonConnector-gateway/src/main/java/com/hms_networks/americas/sc/ignition/comm/responses/Talk2MResponse.java

```diff
@@ -0,0 +1,66 @@
+package com.hms_networks.americas.sc.ignition.comm.responses;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+
+import java.time.Instant;
+import java.time.format.DateTimeParseException;
+import java.util.Date;
+
+/**
+ * An abstract class for responses from a Talk2M API. This class contains any common fields and/or
+ * functionality for all Talk2M responses.
+ *
+ * @author HMS Networks, MU Americas Solution Center
+ * @since 2.0.0
+ * @version 1.0.0
+ */
+public class Talk2MResponse {
+
+  /**
+   * The success status of the logout request.
+   *
+   * @since 1.0.0
+   */
+  private boolean success;
+
+  /**
+   * Gets the success status of the login request.
+   *
+   * @return The success status of the login request.
+   * @since 1.0.0
+   */
+  public boolean getSuccess() {
+    return success;
+  }
+
+  /**
+   * Gets an instance of the specified subclass of {@link Talk2MResponse} from a JSON string.
+   *
+   * @param json The JSON string to parse.
+   * @param clazz The subclass of {@link Talk2MResponse} to parse the JSON string into.
+   * @return An instance of the specified subclass of {@link Talk2MResponse} parsed from the JSON
+   *     string.
+   * @throws JsonSyntaxException if the JSON string is not valid or does not match the specified
+   *     subclass of {@link Talk2MResponse}.
+   * @since 1.0.0
+   */
+  public static <T extends Talk2MResponse> T getFromJson(String json, Class<T> clazz)
```

Comment:
Is the reason for the spelling of clazz just to not exactly match class? 
Seems like if thats the case it could be "destinationClass" or something like that.

Reply:
`class` is a reserved keyword, so you cannot directly name a variable `class`.

Since JDK 1.0, the general Java convention has been to use clazz instead. The JDK itself follows this convention, as you can see instances of `clazz` naming throughout the `Class` class.

Reply:
https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/lang/Class.java

Reply:
okay, thanks for the information. That is a weird choice though.